src/clients/http_client/http_client.ts

```diff
@@ -293,15 +293,13 @@ export function createHttpClientClass(
         ) {
           this.LOGGED_DEPRECATIONS[depHash] = Date.now();
 
-          if (config.logFunction) {
-            const stack = new Error().stack;
-            const log = `API Deprecation Notice ${new Date().toLocaleString()} : ${JSON.stringify(
-              deprecation,
-            )}\n    Stack Trace: ${stack}\n`;
-            await config.logFunction(LogSeverity.Warning, log);
-          } else {
-            console.warn('API Deprecation Notice:', deprecation);
-          }
+          console.warn('API Deprecation Notice:', deprecation);
+
+          const stack = new Error().stack;
+          const log = `API Deprecation Notice ${new Date().toLocaleString()} : ${JSON.stringify(
+            deprecation,
+          )}\n    Stack Trace: ${stack}\n`;
+          await config.logFunction(LogSeverity.Warning, log);
```

Comment:
The previous code did `console.warn` if there was no log function, otherwise the log function was called.  This new code is doing both.  IIRC the original implementation was to reduce the number of warnings (noise) to the console and have them sent to a log somewhere, no?

Reply:
Maybe the default implementation should just be a call to `console.warn`?

Reply:
That's a great callout - maybe we should have a function that just logs for info, warns for warnings, etc... Much cleaner!